src/openvic-simulation/country/CountryInstance.cpp

```diff
@@ -927,7 +939,119 @@ bool CountryInstance::update_rule_set() {
 	return rule_set.trim_and_resolve_conflicts(true);
 }
 
-void CountryInstance::update_gamestate(DefineManager const& define_manager, UnitTypeManager const& unit_type_manager) {
+void CountryInstance::update_modifier_sum(Date today, StaticModifierCache const& static_modifier_cache) {
+	// Update sum of national modifiers
+	base_modifier_sum.clear();
+
+	// Erase expired event modifiers and add non-expired ones to the sum
+	std::erase_if(event_modifiers, [this, today](ModifierInstance const& modifier) -> bool {
+		if (today <= modifier.get_expiry_date()) {
+			base_modifier_sum.add_modifier(*modifier.get_modifier(), this);
+			return false;
+		} else {
+			return true;
+		}
+	});
+
+	// Add static modifiers
+	if (static_modifier_cache.get_base_modifier() != nullptr) {
+		base_modifier_sum.add_modifier(*static_modifier_cache.get_base_modifier(), this);
+	}
+	switch (country_status) {
+		using enum country_status_t;
+	case COUNTRY_STATUS_GREAT_POWER:
+		if (static_modifier_cache.get_great_power() != nullptr) {
+			base_modifier_sum.add_modifier(*static_modifier_cache.get_great_power(), this);
+		}
+		break;
+	case COUNTRY_STATUS_SECONDARY_POWER:
+		if (static_modifier_cache.get_secondary_power() != nullptr) {
+			base_modifier_sum.add_modifier(*static_modifier_cache.get_secondary_power(), this);
+		}
+		break;
+	case COUNTRY_STATUS_CIVILISED:
+		if (static_modifier_cache.get_civilised() != nullptr) {
+			base_modifier_sum.add_modifier(*static_modifier_cache.get_civilised(), this);
+		}
+		break;
+	default:
+		if (static_modifier_cache.get_uncivilised() != nullptr) {
+			base_modifier_sum.add_modifier(*static_modifier_cache.get_uncivilised(), this);
+		}
+	}
+	if (static_modifier_cache.get_disarming() != nullptr && is_disarmed()) {
+		base_modifier_sum.add_modifier(*static_modifier_cache.get_disarming(), this);
+	}
+	if (static_modifier_cache.get_war_exhaustion() != nullptr && war_exhaustion > 0) {
+		base_modifier_sum.add_modifier(*static_modifier_cache.get_war_exhaustion(), this, war_exhaustion);
+	}
+	if (static_modifier_cache.get_infamy() != nullptr && infamy > 0) {
+		base_modifier_sum.add_modifier(*static_modifier_cache.get_infamy(), this, infamy);
+	}
+	if (static_modifier_cache.get_literacy() != nullptr && infamy > 0) {
+		base_modifier_sum.add_modifier(*static_modifier_cache.get_literacy(), this, national_literacy);
+	}
+	if (static_modifier_cache.get_plurality() != nullptr && infamy > 0) {
+		base_modifier_sum.add_modifier(*static_modifier_cache.get_plurality(), this, plurality);
+	}
+	// TODO - difficulty modifiers, war, peace, debt_default_to, bad_debter, generalised_debt_default,
+	//        total_occupation, total_blockaded, in_bankrupcy
+
+	// TODO - handle triggered modifiers
+
+	if (ruling_party != nullptr) {
+		for (Issue const* issue : ruling_party->get_policies()) {
+			if (issue != nullptr) {
```

Comment:
How can issue even be a nullptr here?
Does get_policy return bogus pointers?

Reply:
Party policies are stored in an `IndexedMap` so there's an entry for every `IssueGroup` regardless of whether it's actually set. We'd usually expect parties to have all policies set but it's possible for a mod to miss some in which case there would be a `nullptr` in the policy map.